src/components/mainLayout/youtubeEmbed/youtubeEmbed.tsx

```diff
@@ -0,0 +1,36 @@
+import { useState } from "react";
```

Comment:
you need to add "use client" in the first line for using hooks in server components